tensorflow/tools/ci_build/builds/test_on_install.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# Build the Python PIP installation package for TensorFlow
+# and run the Python unit tests from the source code on the installation
+#
+# When executing the Python unit tests, the script obeys three environment
+# variables: PY_TEST_WHITELIST, PY_TEST_BLACKLIST and PY_TEST_GPU_BLACKLIST
+#
+# To select only a subset of the Python tests to run, set the environment
+# variable PY_TEST_WHITELIST, e.g.,
+#   PY_TEST_WHITELIST="tensorflow/python/kernel_tests/shape_ops_test.py"
+# Separate the tests with a colon (:). Leave this environment variable empty
+# to disable the whitelist.
+#
+# You can also ignore a set of the tests by using the environment variable
+# PY_TEST_BLACKLIST. For example, you can include in PY_TEST_BLACKLIST the
+# tests that depend on Python modules in TensorFlow source that are not
+# exported publicly.
+#
+# In addition, you can put blacklist for only GPU build inthe environment
+# variable PY_TEST_GPU_BLACKLIST.
+#
+
+echo "PY_TEST_BLACKLIST: ${PY_TEST_BLACKLIST}"
```

Comment:
In total, there are five tests that failed for the CPU PIP install test:
1) tensorflow/python/tools/freeze_graph_test.py
2) tensorflow/python/ops/image_ops_test.py
3) tensorflow/python/framework/ops_test.py
4) tensorflow/python/util/protobuf/compare_test.py
5) tensorflow/python/framework/device_test.py

1 is fixed by copying the tensorflow/python/tools folder, which is not included in the installation by default, to the PIP installation destination package.
2 is fixed by copying some JPEG and PNG images for test to the working directory for testing. 

I added the fixes for 1 and 2 to test_on_install.sh. 

3-5 cannot be fixed easily. 3 depends on "test_ops", which is defined in a C++ file wrapped as a .py file through the Bazel rule “tf_gen_ops_wrapper_py”. 4 depends on a similar thing defined outside Python (compare_test_pb2). 5 depends on external things as well (CheckValid and ToString).

I do think having the capability to skip some tests will come in handy.